api/envoy/config/route/v3/route_components.proto

```diff
@@ -1362,6 +1362,24 @@ message RouteAction {
   MaxStreamDuration max_stream_duration = 36;
 }
 
+// Holds the segments for rewriting urls base on pattern templates
+message RouteUrlRewritePattern {
```

Comment:
Could you provide an example of `RouteUrlRewritePattern`? And we have `envoy::type::matcher::v3::RegexMatchAndSubstitute` currently, can we extend it if it does not meet your need?

Reply:
Why is this proto in the public API if not referenced anywhere in it?

Reply:
Far enough. Original use before port is gone now... Let me try that.